### PR TITLE
#23 Refacto InputColumn code before implementing cleaning scripts

### PR DESCRIPTION
**Pull request by Nabellaleen**
Juste a refacto - without any functionnal change, to split the code of the InputColumn in smaller chunks

### CODE REVIEW

File: client/src/views/mapping/components/inputColumn/addJoinButton.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { Button } from "@blueprintjs/core";
+
+import * as React from "react";
```
Comment (simonvadee) [inline client/src/views/mapping/components/inputColumn/addJoinButton.tsx]: i prefer importing only the symbols needed from react

File: client/src/views/mapping/components/inputColumn/index.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import { Button, Card, Elevation, Tag } from "@blueprintjs/core";
+import * as React from "react";
+import { Subscription } from "react-apollo";
+
+import { ISelectedSource } from "../../../../types";
```
Comment (simonvadee) [inline client/src/views/mapping/components/inputColumn/index.tsx]: can you use absolute imports instead ?

File: client/src/views/mapping/components/inputColumn/index.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import { Button, Card, Elevation, Tag } from "@blueprintjs/core";
+import * as React from "react";
```
Comment (simonvadee) [inline client/src/views/mapping/components/inputColumn/index.tsx]: same as before

File: client/src/views/mapping/components/inputColumn/join.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import { Button } from "@blueprintjs/core";
+import * as React from "react";
+import { Mutation, Subscription } from "react-apollo";
+
+import { ISelectedSource } from "../../../../types";
```
Comment (simonvadee) [inline client/src/views/mapping/components/inputColumn/join.tsx]: absolute imports

File: client/src/views/mapping/components/inputColumn/joinColumns.tsx [hunk 5]
```diff
@@ -1,10 +1,9 @@
 import * as React from "react";
-import { Mutation, Query, Subscription } from "react-apollo";
 
-import { ISelectedSource } from "../../../types";
+import { ISelectedSource } from "../../../../types";
```
Comment (simonvadee) [inline client/src/views/mapping/components/inputColumn/joinColumns.tsx]: same

File: client/src/views/mapping/components/inputColumn/joinColumns.tsx [hunk 6]
```diff
@@ -1,10 +1,9 @@
...
 
 // COMPONENTS
-import ColumnPicker from "../../../components/columnPicker";
+import ColumnPicker from "../../../../components/columnPicker";
```
Comment (simonvadee) [inline client/src/views/mapping/components/inputColumn/joinColumns.tsx]: same

File: client/src/views/mapping/components/inputColumn/scriptTag.tsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import { Tag } from "@blueprintjs/core";
+
+import * as React from "react";
+
+import { Mutation } from "react-apollo";
+
+import StringSelect from "../../../../components/selects/stringSelect";
```
Comment (simonvadee) [inline client/src/views/mapping/components/inputColumn/scriptTag.tsx]: same

File: client/src/views/mapping/components/inputColumn/addJoinButton.tsx (hunk 1 above)
Comment (Nabellaleen) [reply client/src/views/mapping/components/inputColumn/addJoinButton.tsx]: After some checks, this import is mandatory.
If missing, the TSX compiler make this error : `TS2686: 'React' refers to a UMD global, but the current file is a module. Consider adding an import instead.`

File: client/src/views/mapping/components/inputColumn/index.tsx (hunk 2 above)
Comment (Nabellaleen) [reply client/src/views/mapping/components/inputColumn/index.tsx]: Ok, je rajoute ce qu'il faut dans le tsconfig pour le faire

File: client/src/views/mapping/components/inputColumn/addJoinButton.tsx (hunk 1 above)
Comment (alexis-thual) [reply client/src/views/mapping/components/inputColumn/addJoinButton.tsx]: I think if you merge from master, this error should now go away.

File: client/src/views/mapping/components/inputColumn/addJoinButton.tsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+import { Mutation } from "react-apollo";
+
+const createJoinAndUpdateInputColumn = require("../../../../graphql/mutations/createJoinAndUpdateInputColumn.graphql");
```
Comment (alexis-thual) [inline client/src/views/mapping/components/inputColumn/addJoinButton.tsx]: If you merge from master, you can also use absolute imports (here, `src/graphql/......` should work)

File: client/src/views/mapping/components/inputColumn/addJoinButton.tsx [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+interface IProps {
+  column: any;
+}
+
+const InputColumnAddJoin = ({ column }: IProps) => (
+  <Mutation mutation={createJoinAndUpdateInputColumn}>
```
Comment (alexis-thual) [inline client/src/views/mapping/components/inputColumn/addJoinButton.tsx]: Although I am not doing it yet, I think I'll stop using `Query` and `Mutation` tags, and rather opt for `useQuery` or `useMutation` from `@apollo/react-hooks`: https://www.apollographql.com/docs/react/api/react-hooks/#usequery

Imho, it prevents nesting "service logic" and "components" and leads to cleaner code :slightly_smiling_face: 
Don't necessarily change this component now though...

File: client/src/views/mapping/components/inputColumn/breadcrumbs.tsx [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+      <Tag minimal={true}>{label.toUpperCase()}</Tag>
+      <Tag intent={"success"} large={true}>
+        {value}
+      </Tag>
+    </div>
+  );
+}
+
+const InputColumnBreadcrumbs = ({ source, column }: IProps) => {
+  const breadcrumbTags = [];
```
Comment (alexis-thual) [inline client/src/views/mapping/components/inputColumn/breadcrumbs.tsx]: I would rewrite this using React hooks:
```javascript
const [breadcrumbTags, setBreadcrumbTags] = React.useState([]);
React.useEffect(() => {
let breadcrumbTags = [];
if (source.hasOwner) {
    breadcrumbTags.push({
      text: buildReadOnlyBreadcrumbTag("owner", column.owner)
    });
  }
  breadcrumbTags.push(
    { text: buildReadOnlyBreadcrumbTag("table", column.table) },
    { text: buildReadOnlyBreadcrumbTag("column", column.column) }
  );
  setBreadcrumbTags(breadcrumbTags);
}, [source]);
```

**Pull request closed without merging**